File: server/src/Models/workModel.js
import mongoose from "mongoose";

const workSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
    },
    location: {
      type: String,
      required: true,
    },
    cno: {
      type: Number,
      required: true,
      unique: true,
    },
    startDate: {
      type: Date,
      required: true,
    },
    endDate: {
      type: Date,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    managerIds: {
      type: [mongoose.Schema.Types.ObjectId],
      ref: "User",
    },
  },
  { timestamps: true }
);

workSchema.statics.isUnique = async function (cno) {
  try {
    if (!cno) throw new Error("Dados inválidos");

    const cnoInDatabase = await this.findOne({ cno });

    if (cnoInDatabase) return false;
    return true;
  } catch (err) {
    console.log("Erro ao verificar dados", err.message);
    return false;
  }
};

export default mongoose.model("Work", workSchema);

File: server/src/models/workModel.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Work struct {
	ID         primitive.ObjectID   `bson:"_id,omitempty" json:"id"`
	Name       string               `bson:"name" json:"name" validate:"required"`
	Location   string               `bson:"location" json:"location" validate:"required"`
	CNO        int                  `bson:"cno" json:"cno" validate:"required"`
	StartDate  time.Time            `bson:"startDate,omitempty" json:"startDate,omitempty"`
	EndDate    time.Time            `bson:"endDate,omitempty" json:"endDate,omitempty"`
	IsActive   bool                 `bson:"isActive" json:"isActive"`
	ManagerIDs []primitive.ObjectID `bson:"managerIds,omitempty" json:"managerIds,omitempty"`
}
